build/gulp/tasks/bundle.ts

```diff
@@ -2,70 +2,79 @@ import { task, series, parallel, src, dest } from 'gulp'
 import * as merge2 from 'merge2'
 import * as rimraf from 'rimraf'
 import * as webpack from 'webpack'
+import { argv } from 'yargs'
 
 import config from '../../../config'
+import sh from '../sh'
 
 const { paths } = config
 const g = require('gulp-load-plugins')()
 const { log, PluginError } = g.util
 
+const buildUMD = !!argv.umd
+const packageName: string = argv.package
+
 // ----------------------------------------
 // Clean
 // ----------------------------------------
 
-task('clean:dist:es', cb => {
-  rimraf(`${config.paths.dist()}/es/*`, cb)
+task('bundle:clean:es', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/es/*`, cb)
 })
 
-task('clean:dist:commonjs', cb => {
-  rimraf(`${config.paths.dist()}/commonjs/*`, cb)
+task('bundle:clean:commonjs', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/commonjs/*`, cb)
 })
 
-task('clean:dist:umd', cb => {
-  rimraf(`${config.paths.dist()}/umd/*`, cb)
+task('bundle:clean:umd', cb => {
+  rimraf(`${config.paths.packageDist(packageName)}/umd/*`, cb)
 })
 
-task('clean:dist', parallel('clean:dist:es', 'clean:dist:commonjs', 'clean:dist:umd'))
+task('bundle:clean', parallel('bundle:clean:es', 'bundle:clean:commonjs', 'bundle:clean:umd'))
 
 // ----------------------------------------
 // Build
 // ----------------------------------------
-const componentsSrc = [paths.src('**/*.{ts,tsx}'), `!${paths.src('**/umd.ts')}`]
+const createComponentSrc = () => [
```

Comment:
why delayed execution necessary to get this array?

Reply:
If we will execute just `yarn gulp`, the `packageName` variable will undefined and we will get an error here.

---

We can revert this and modify code:
```diff
-const packageName: string = argv.package
+const packageName: string = argv.package || ''
```